cli/icgcget/commands/reports.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2016 The Ontario Institute for Cancer Research. All rights reserved.
+#
+# This program and the accompanying materials are made available under the terms of the GNU Public License v3.0.
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+# OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+# SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+# TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+import logging
+import click
+import os
+from collections import OrderedDict
+from cli.icgcget.clients.ega.ega_client import EgaDownloadClient
+from cli.icgcget.clients.errors import SubprocessError
+from cli.icgcget.clients.gdc.gdc_client import GdcDownloadClient
+from cli.icgcget.clients.icgc.storage_client import StorageClient
+from cli.icgcget.clients.pdc.pdc_client import PdcDownloadClient
+from cli.icgcget.clients.gnos.gnos_client import GnosDownloadClient
+from cli.icgcget.clients.utils import convert_size, donor_addition, increment_types, build_table
+
+from tabulate import tabulate
+
+from utils import check_access, api_error_catch,  get_entities
+
+
+class StatusScreenDispatcher:
+    def __init__(self):
+        self.logger = logging.getLogger("__log__")
+        self.gdc_client = GdcDownloadClient()
```

Comment:
I think we should use dependency injection here. The constructor should accept a dictionary of clients. Then all commands can do:

```clients.ega.download()```

Either that or define all as fields in the base class and then call super. The main point is to remove the constructor call so that you can mock them in tests, etc.